packages/web/src/new/services/LocationService/LocationService.ts

```diff
@@ -1,68 +1,61 @@
-import { matchPath } from 'react-router';
+import type { Location, NavigateFunction, Params, Path } from 'react-router-dom';
+import { matchPath } from 'react-router-dom';
 
-import type { History, Location, LocationState, Path, UnregisterCallback } from 'history';
 import { action, makeObservable, observable } from 'mobx';
 import assert from 'ts-invariant';
 import { singleton } from 'tsyringe';
 
 @singleton()
 export class LocationService {
-  _location?: Location;
-
-  private _history?: History;
-  private _removeListener?: UnregisterCallback;
+  private _location?: Location;
+  private _navigate?: NavigateFunction;
 
   constructor() {
-    makeObservable(this, {
+    makeObservable<LocationService, '_location'>(this, {
       _location: observable,
+
+      setLocation: action,
     });
   }
 
-  private _assertHistory(): void {
-    assert(this._history, 'History is not set in LocationService');
+  private _assertNavigate(): void {
+    assert(this._navigate, 'NavigateFunction is not set in LocationService');
   }
 
   private _assertLocation(): void {
     assert(this._location, 'Location is not set in LocationService');
   }
 
-  private _listenHistory(): void {
-    if (this._removeListener) {
-      this._removeListener();
-    }
-
-    this._assertHistory();
-    this._removeListener = this._history!.listen(
-      action((location) => {
-        this._location = location;
-      }),
-    );
+  setLocation(location: Location): void {
+    this._location = location;
   }
 
-  setHistory(history: History): void {
-    this._history = history;
-
-    this._listenHistory();
+  setNavigate(navigate: NavigateFunction): void {
+    this._navigate = navigate;
   }
 
-  getParams<Params>(pathTemplate: string): Params {
+  getParams<ParamKey extends string>(pathTemplate: string): Params<ParamKey> {
```

Comment:
If we have pathTemplate with few params and we need to get few params we will need to call getParams twice. so is it a better realization than it was before?

Reply:
TS shows error otherwise.
In order to have several params we can set them by using union: 'symbol' | 'publicKey'